datafusion/optimizer/src/optimizer.rs

```diff
@@ -80,12 +84,31 @@ pub trait OptimizerRule {
     /// A human readable name for this optimizer rule
     fn name(&self) -> &str;
 
-    /// How should the rule be applied by the optimizer? See comments on [`ApplyOrder`] for details.
+    /// How should the rule be applied by the optimizer? See comments on
+    /// [`ApplyOrder`] for details.
     ///
-    /// If a rule use default None, it should traverse recursively plan inside itself
+    /// If returns `None`, the default, the rule must handle recursion itself
     fn apply_order(&self) -> Option<ApplyOrder> {
         None
```

Comment:
I think most of the cases are BottomUp by default?

Reply:
> I think most of the cases are BottomUp by default?

we can't assume it.

Reply:
I agree most of the cases probably *should be* bottom up (but not all). I'll keep an eye open for this as I work through the other optimizer rules